File: cpp/LongestSubstringWithoutRepeatingCharacters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.size()==0)
            return 0;
        int max_len=1;
        string temp=s.substr(0,1);
        cout<<temp<<endl;
        for(int i=1;i<s.size();++i){
            if(temp.find(s[i])==string::npos){
                temp+=s[i];
            }
            else{
                if(max_len<temp.size())
                    max_len=temp.size();
                temp=temp.substr(temp.find(s[i])+1,temp.size())+s[i];
            }
        }
        if(max_len<temp.size())
            max_len=temp.size();
        return max_len;
    }
};

File: java/LongestSubstringWithoutRepeatingCharacters.java
public class Solution {  
    public int lengthOfLongestSubstring(String s) {  
        if(s.length()<=1){  
            return s.length();  
        }  
        String temp = s.substring(0,1);  
        int maxlength = 1;  
  
        for(int i=1;i<s.length();i++){  
            if(!temp.contains(String.valueOf(s.charAt(i)))){  
                temp += s.charAt(i);  
            }
            else{  
                if(maxlength<temp.length()){  
                    maxlength = temp.length();  
                }  
                temp = temp.substring(temp.indexOf(s.charAt(i))+1) + String.valueOf(s.charAt(i));  
            }  
        }  

        if(maxlength<temp.length()){
            maxlength = temp.length(); 
        }
        return maxlength;  
    } 

}  
